### DISCUSSION TITLE
'no applicable method 'Parse' exists in type 'Double''

### DISCUSSION THREAD

**Discussion by chiyu1203** [bonsai-rx/bonsai#958] [Q&A]
Dear Bonsai Developers,

I followed BonVision's documents <https://bonvision.github.io/pages/08-GratingSeries/#step-2-enter-parameters-of-stimuli> to create some gratings stimuli. Adopting the workflow from B. Alternate Method: Loading Parameters from a CSV file, I attempted to use double.Parse in the ExpressionTransform. However, Bonsai complained that 'no applicable method 'Parse' exists in type 'Double''
I have tried the same working environment with another Windows 10 computer but still bumped into the same error.
Could this be a bug that is introduced rather lately?? 
Below are the configuration of my working environment:
Windows 10
Bonsai 2.6.3
Packages:
Default packages of Bonsai 2.6.3
+ the required packages from BonVision website 
(Starter Pack
Numerics
WindowInput Library
BonVision)

**Comment by bruno-f-cruz** [comment]
Not sure if this will fix it. but I think you are not passing the input to the method. Try:
`Double.Parse(it)`

![image](https://user-images.githubusercontent.com/7049351/177013596-26f8284b-b261-4d64-b879-f31784a67132.png)

**Reply by chiyu1203** [reply]
Hi Bruno, I have tried that expression and it works but still if I switch back to the Expression from BonVision website, I saw the same error. Is it because they tried to add a new() to bracket the whole thing? then do I need to add some custom extension?
![GratingSeriesB1-ExpTrans2](https://user-images.githubusercontent.com/16696523/177118803-0108cdec-8b00-475d-a21a-d000d50b1899.png)

**Reply by bruno-f-cruz** [reply]
Can you include a workflow? thanks

**Reply by chiyu1203** [reply]
Hi Bruno,

Here is the workflow. Thank you!
[drifting_grating_3tf.bonsai.zip](https://github.com/bonsai-rx/bonsai/files/9055004/drifting_grating_3tf.bonsai.zip)

BEST REGARDS

Chi-Yu

**Reply by bruno-f-cruz** [reply]
You are passing a `char` type instead of a `string` one. To fix it, either add a valid separator to your `CsvReader` node (e.g. `,`) which will output a `string[]`, or, if you really want to parse a single character in the incoming string, use the `char.ToString()` method (e.g. Double.Parse(it[0].ToString()).

Hope it helps!

**Reply by chiyu1203** [reply]
Hi Bruno,

Yes, it works!! Thank you so much! 
I should have noticed this!!

Best Regards

Chi-Yu
